### PR TITLE
Add EKS AWSManagedMachinePool example to spot instances doc

### LINKED ISSUE #2523: Spot Instance Support for AWSManagedMachinePool
/kind feature

**Describe the solution you'd like**
I'd like to use AWSManagedMachinePool / MachinePool with spot instances.

**Environment:**

- Cluster-api-provider-aws version: v6.5.0
- Kubernetes version: (use `kubectl version`): v1.19
- OS (e.g. from `/etc/os-release`): amazon linux 2

### PR DESCRIPTION
**Pull request by pydctw**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add an example how to use spot instances with a EKS managed node group backed by AWSManagedMachinePool.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2523 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [x] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: docs/book/src/topics/spot-instances.md [hunk 1]
```diff
@@ -34,4 +34,17 @@ spec:
     spotMarketOptions:
       maxPrice: 0.02 # Price in USD per hour (up to 5 decimal places)
 ```
-> **IMPORTANT NOTE**: The experimental feature `MachinePool` does not support using spot instances as of now.
```
Comment (sedefsavas) [inline docs/book/src/topics/spot-instances.md:37]: This is correct, good to have a note or warning on this here if there is no other warning in the doc.

Comment (Ankitasw) [reply docs/book/src/topics/spot-instances.md:37]: +1

Comment (pydctw) [reply docs/book/src/topics/spot-instances.md:37]: A good point. Slightly modified the note from `MachinePool` to `AWSMachinePool` as one variation of machine pool implementation, AWSManagedMachinePool, supports spot instance.

**Pull request merged**